libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -1860,6 +1867,61 @@ namespace
    }
 }
 
+auto WaveTrack::FindWideClip(const WaveClip &clip, int *pDistance) const
+   -> ConstIterPair
+{
+   const WaveClipHolders &list = mClips;
+   const WaveClipHolders *pList2{};
+   const auto channels = TrackList::Channels(this);
+   if (channels.size() > 1)
+      pList2 = &(*channels.rbegin())->mClips;
+   int distance = 0;
+   auto it = list.begin();
+   for (const auto end = list.end(); it != end; ++it) {
+      if (it->get() == &clip)
+         break;
+      ++distance;
+   }
+   if (pDistance)
+      *pDistance = distance;
+   WaveClipHolders::const_iterator it2{};
+   if (pList2)
+      it2 = pList2->begin() + distance;
+   return { it, it2 };
+}
+
+auto WaveTrack::FindWideClip(const WaveClip &clip, int *pDistance)
+   -> IterPair
+{
+   WaveClipHolders &list = mClips;
+   WaveClipHolders *pList2{};
+   const auto channels = TrackList::Channels(this);
+   if (channels.size() > 1)
+      pList2 = &(*channels.rbegin())->mClips;
+   int distance = 0;
+   auto it = list.begin();
+   for (const auto end = list.end(); it != end; ++it) {
+      if (it->get() == &clip)
+         break;
+      ++distance;
+   }
+   if (pDistance)
+      *pDistance = distance;
+   WaveClipHolders::iterator it2{};
+   if (pList2)
+      it2 = pList2->begin() + distance;
+   return { it, it2 };
+}
+
+void WaveTrack::RemoveWideClip(IterPair pair)
+{
+   mClips.erase(pair.first);
+   const auto channels = TrackList::Channels(this);
+   if (channels.size() > 1)
+      (*channels.rbegin())->mClips.erase(pair.second);
```

Comment:
This assumes that `pair.second` really is an iterator obtained from the same leader-follower instance. Safer would be to modify the integration of the parts such that `RemoveWideClip(*clip->GetClip(0));` can be written (instead of current `RemoveWideClip(FindWideClip(*clip->GetClip(0)));`).

Reply:
Both new functions are `private` so there is a bounded amount of code to examine to ensure the correctness.  I wouldn't worry about it.